src/integer_mod_q/mat_zq/set.rs

```diff
@@ -161,25 +164,17 @@ impl MatZq {
     ) -> Result<(), MathError> {
         let col0 = evaluate_index(col0)?;
         let col1 = evaluate_index(col1)?;
-        if col0 >= self.get_num_columns() {
-            return Err(MathError::OutOfBounds(
-                format!("smaller than {}", self.get_num_columns()),
-                col0.to_string(),
-            ));
-        }
-        if col1 >= other.get_num_columns() {
-            return Err(MathError::OutOfBounds(
-                format!("smaller than {}", other.get_num_columns()),
-                col1.to_string(),
-            ));
-        }
-        if self.get_num_rows() != other.get_num_rows() {
-            return Err(MathError::MismatchingMatrixDimension(format!(
-                "as set_column was called on two matrices with different number of rows {} and {}",
-                self.get_num_rows(),
-                other.get_num_rows()
-            )));
-        }
+
+        evaluate_matrix_set_row(
+            "set_column",
+            col0,
+            col1,
+            self.get_num_columns(),
+            other.get_num_columns(),
+            self.get_num_rows(),
+            other.get_num_rows(),
```

Comment:
here you provide column numbers to a function that has parameters row, that is not intuitive